qa/workunits/rbd/test_admin_socket.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash -ex
+
+TMPDIR=/tmp/rbd_test_admin_socket$$
+mkdir $TMPDIR
+trap "rm -fr $TMPDIR" 0
+
+. $(dirname $0)/../ceph-helpers.sh
+
+function expect_false()
+{
+    set -x
+    if "$@"; then return 1; else return 0; fi
+}
+
+function rbd_watch_out_file()
+{
+    echo ${TMPDIR}/rbd_watch_$1.out
+}
+
+function rbd_watch_pid_file()
+{
+    echo ${TMPDIR}/rbd_watch_$1.pid
+}
+
+function rbd_watch_fifo()
+{
+    echo ${TMPDIR}/rbd_watch_$1.fifo
+}
+
+function rbd_watch_asok()
+{
+    echo ${TMPDIR}/rbd_watch_$1.asok
+}
+
+function rbd_get_perfcounter()
+{
+    local image=$1
+    local counter=$2
+
+    ceph --admin-daemon $(rbd_watch_asok ${image}) perf dump |
+	jq ".\"librbd--rbd/${image}\".${counter}"
```

Comment:
As for your comment that jq is not necessarily installed, we have in  run-make-check.sh the instruction to install jq, before running the tests. That is why I decided it would be ok to use jq here, and not parse this e.g. with awk, which is fragile. I could also use a python oneliner with json module, which could be correct but heavier.